cs3api4lab/api/lock_manager.py

```diff
@@ -74,10 +75,10 @@ def resolve_file_path(self, file_path, endpoint):
         return file_path
 
     def _resolve_directory(self, dir_path, endpoint):#right now its possible to write in somone else's directory without it being shared
-        try:
-            self.storage_logic.stat(dir_path, endpoint)
+        stat = self.storage_api.stat(dir_path, endpoint)
+        if stat.status.code == cs3code.CODE_OK:
             return dir_path
-        except Exception:
+        else:
             return self.config.mount_dir + '/'
```

Comment:
This might lead to conflicts between files with the same name. I was going to ask if we can de-duplicate, but since we will switch to the lock api, doesn't seem worth the trouble.